app/src/main/java/com/github/se/polyfit/ui/screen/IngredientScreen.kt

```diff
@@ -0,0 +1,214 @@
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.compose.rememberNavController
+import com.github.se.polyfit.ui.components.GradientButton
+import com.github.se.polyfit.ui.navigation.Navigation
+import com.github.se.polyfit.ui.theme.PrimaryPurple
+
+// TODO: Replace with real data class
+data class Ingredient(
+    val name: String,
+    val quantity: Int,
+    val unit: String,
+    val probability: Float = 1.0f,
+) {}
+
+@Composable
+fun IngredientScreen(
+    navigation: Navigation,
+    initialIngredients: List<Ingredient>,
+    initialPotentialIngredients: List<Ingredient>,
+    // TODO:  ingredientsViewModel: IngredientsViewModel = IngredientsViewModel(),
+) {
+
+  val ingredients = remember { mutableStateListOf(*initialIngredients.toTypedArray()) }
+  val potentialIngredients = remember {
+    mutableStateListOf(*initialPotentialIngredients.toTypedArray())
+  }
+
+  Scaffold(topBar = { TopBar(navigation) }, bottomBar = { BottomBar() }) {
+    IngredientList(
+        it,
+        ingredients,
+        potentialIngredients,
+        onAddIngredient = { index ->
+          val item = potentialIngredients.removeAt(index)
+          ingredients.add(item)
+        })
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun IngredientList(
+    it: PaddingValues,
+    ingredients: List<Ingredient>,
+    potentialIngredients: List<Ingredient>,
+    onAddIngredient: (Int) -> Unit,
+) {
+
+  val initialSuggestions = 3
+  val potentialIndex = remember {
+    // equivalent to min(3, potentialIngredients.size)
+    mutableIntStateOf(initialSuggestions.coerceAtMost(potentialIngredients.size))
+  }
+  Column(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(it)
+              .verticalScroll(rememberScrollState())
+              .testTag("IngredientsList"),
+      verticalArrangement = Arrangement.Top,
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        if (ingredients.isEmpty() && potentialIngredients.isEmpty()) {
+          Text(
+              "No ingredients added yet",
+              modifier = Modifier.fillMaxSize().padding(16.dp, 0.dp).testTag("NoIngredients"),
+              color = MaterialTheme.colorScheme.secondary,
+              fontSize = MaterialTheme.typography.headlineSmall.fontSize)
+        } else {
+          FlowRow(
+              horizontalArrangement = Arrangement.Start,
+              verticalArrangement = Arrangement.Top,
+              maxItemsInEachRow = Int.MAX_VALUE) {
```

Comment:
This is already the default value, so it is not needed.

```suggestion
              ) {
```